ext/standard/crc32.c

```diff
@@ -116,9 +104,47 @@ PHP_FUNCTION(crc32)
 	p += nr_simd;
 #endif
 
+	/* The trailing part */
 	for (; nr--; ++p) {
 		crc = ((crc >> 8) & 0x00FFFFFF) ^ crc32tab[(crc ^ (*p)) & 0xFF ];
 	}
+
+	return crc;
+}
+
+uint32_t crc32_stream_bulk_update(uint32_t crc, php_stream *fp, size_t nr)
+{
+	size_t handled = 0, n;
+	char buf[1024];
+
+	while (handled < nr) {
+		n = nr - handled;
+		n = (n < sizeof(buf)) ? n : sizeof(buf); /* tweak to buf size */
+
+		php_stream_read(fp, buf, n);
```

Comment:
If this fails, wouldn't `buf` contain the previous read and possibly garbage depending on the value of `n`? Perhaps should just break if `EOF`?

thanks.

Reply:
> If this fails, wouldn't `buf` contain the previous read and possibly garbage depending on the value of `n`? Perhaps should just break if `EOF`?
> 
> thanks.

I change this API to return `FAILUR` for `EOF`. Caller can check return value for this corner case.